dashboard/app/models/user.rb

```diff
@@ -374,9 +374,24 @@ def hash_email
     self.hashed_email = User.hash_email(email)
   end
 
-  def sanitize_race_data
+  # Returns whether the comma separated list of races represents an under-represented minority user.
+  # @return [Boolean, nil] Whether races_comma_separated represents a URM user.
+  #   - true: Yes, a URM user.
+  #   - false: No, not a URM user.
+  #   - nil: Don't know, may or may not be a URM user.
+  def self.urm_from_races(races_comma_separated)
+    races_array = races_comma_separated.split(',')
+    return nil if races_array.empty?
+    return nil if (races_array & ['opt_out', 'nonsense', 'closed_dialog']).any?
+    return true if (races_array & ['black', 'hispanic', 'hawaiian', 'american_indian']).any?
+    false
+  end
+
+  def sanitize_and_set_race_data
     return unless property_changed?('races')
+    return unless races
```

Comment:
Note that, because of overloaded methods, both `property_changed('races')` and `races` appear to refer to the SerializedAttribute key/value pair, not the column. That said, let me fix this to be less hacky.